include/zephyr/net/net_event.h

```diff
@@ -257,6 +257,19 @@ struct net_event_ipv6_route {
 	uint8_t prefix_len;
 };
 
+/**
+ * @brief Network Management event information structure
+ * Used to pass information on network events like
+ *   NET_EVENT_IPV6_PREFIX_ADD and
+ *   NET_EVENT_IPV6_PREFIX_DEL
+ * when CONFIG_NET_MGMT_EVENT_INFO is enabled and event generator pass the
+ * information.
+ */
+struct net_event_ipv6_prefix {
+	struct in6_addr addr; /* prefix */
+	uint8_t len;
+	uint32_t lifetime;
+};
 #endif /* CONFIG_NET_MGMT_EVENT_INFO */
```

Comment:
It seems that hiding the struct definitions behind `#ifdef CONFIG_NET_MGMT_EVENT_INFO` causes issues when `IS_ENABLED()` is used in the code:
> /__w/zephyr/zephyr/subsys/net/ip/net_if.c:2457:54: error: storage size of 'info' isn't known
 2457 |                         struct net_event_ipv6_prefix info;
 
 I wonder if that's actually needed though, given that the header is just defining new types here. I think we should just get rid of that #ifdef, WDYT @jukkar?

Reply:
@rlubos ack to that